src/OfflineTool.cpp

```diff
@@ -0,0 +1,473 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of ripple-offline-tool:
+        https://github.com/ximinez/ripple-offline-tool
+    Copyright (c) 2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <OfflineTool.h>
+#include <RippleKey.h>
+#include <Serialize.h>
+#include <ripple/beast/core/SemanticVersion.h>
+#include <ripple/beast/unit_test.h>
+#include <beast/unit_test/dstream.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/format.hpp>
+#include <boost/program_options.hpp>
+
+//------------------------------------------------------------------------------
+char const* const versionString =
+
+    //--------------------------------------------------------------------------
+    //  The build version number. You must edit this for each release
+    //  and follow the format described at http://semver.org/
+    //
+        "0.1.0"
+
+#if defined(DEBUG) || defined(SANITIZER)
+       "+"
+#ifdef DEBUG
+        "DEBUG"
+#ifdef SANITIZER
+        "."
+#endif
+#endif
+
+#ifdef SANITIZER
+        BEAST_PP_STR1_(SANITIZER)
+#endif
+#endif
+
+    //--------------------------------------------------------------------------
+    ;
+
+static
+int
+runUnitTests ()
+{
+    using namespace beast::unit_test;
+    beast::unit_test::dstream dout{std::cout};
+    reporter r{dout};
+    bool const anyFailed = r.run_each(global_suites());
+    if(anyFailed)
+        return EXIT_FAILURE;    //LCOV_EXCL_LINE
+    return EXIT_SUCCESS;
+}
+
+void
+doSerialize(std::string const& data)
+{
+    auto const tx = [&]
+    {
+        auto const json = serialize::parseJson(data);
+        return json ? serialize::makeObject(json) : boost::none;
+    }();
+    if (!tx)
+    {
+        std::cout << "Unable to serialize \"" << data << "\"" <<
```

Comment:
Since we allow serialization and deserialization from `stdin`, I wonder if error message text ought to be sent to `cerr`, rather than `cout`?